components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/Oauth2ScopeUtils.java

```diff
@@ -321,4 +324,44 @@ public static boolean isSystemLevelInternalSystemScopeManagementEnabled() {
         }
         return true;
     }
+
+    /**
+     * Iterate through the scopes array to filter out the internal scopes.
+     * @param scopes String array of scopes.
+     * @return String array with internal scopes. Return an empty array if there's not any internal scopes in the
+     * given scopes array.
+     */
+    public static String[] getRequestedScopes(String[] scopes) {
+
+        List<String> requestedScopes = new ArrayList<>();
+        if (scopes == null) {
+            return null;
+        }
+        for (String scope : scopes) {
+            if (scope.startsWith(INTERNAL_SCOPE_PREFIX) || scope.equalsIgnoreCase(SYSTEM_SCOPE)) {
+                requestedScopes.add(scope);
+            }
+        }
+        return requestedScopes.toArray(new String[0]);
+    }
+
+    /**
+     * Iterate through the scopes list to filter out the internal scopes.
+     * @param scopes String list of scopes.
+     * @return String list with internal scopes. Return an empty list if there's not any internal scopes in the
+     * given scopes list.
+     */
+    public static List<String> getRequestedScopes(List<String> scopes) {
+
+        List<String> requestedScopes = new ArrayList<>();
+        if (scopes == null) {
+            return null;
```

Comment:
Let's not return null.
Instead `return Collections.emptyList();`

Reply:
This has been fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1766/commits/ca0b0b94f26a3d0f66fc9d3e6b0ba33c7dd8075b